File: java/0412.0_Fizz_Buzz.java
/*
执行用时：2 ms, 在所有 Java 提交中击败了49.78% 的用户
内存消耗：39.6 MB, 在所有 Java 提交中击败了56.25% 的用户
通过测试用例：8 / 8
*/

class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> ans = new ArrayList<>();
        for (Integer i = 1; i <= n; i++) {
            if (i % 15 == 0) ans.add("FizzBuzz");
            else if (i % 3 == 0) ans.add("Fizz");
            else if (i % 5 == 0) ans.add("Buzz");
            else ans.add(i.toString());
        }
        return ans;
    }
}

File: leetcode-cn/0412.0_Fizz_Buzz.py
'''
执行用时：36 ms, 在所有 Python3 提交中击败了45.86% 的用户
内存消耗：15.4 MB, 在所有 Python3 提交中击败了20.41% 的用户
通过测试用例：8 / 8
'''
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        ans = []
        for i in range(n):
            if (i + 1) % 15 == 0:
                ans.append('FizzBuzz')
            elif (i + 1) % 3 == 0:
                ans.append('Fizz')
            elif (i + 1) % 5 == 0:
                ans.append('Buzz')
            else:
                ans.append(str(i + 1))
        return ans

